sessiontxn/staleread/processor.go

```diff
@@ -90,6 +90,21 @@ func (p *baseProcessor) OnExecutePrepared(_ StalenessTSEvaluator) error {
 	return errors.New("not supported")
 }
 
+func (p *baseProcessor) evaluateFromExternalTS() error {
+	// if stale read is not used explicitly, check whether we should use external timestamp
+	if p.sctx.GetSessionVars().EnableExternalTSRead {
+		externalTimestamp, err := GetExternalTimestamp(p.ctx, p.sctx)
+		if err != nil {
+			return errAsOf.FastGenWithCause(err.Error())
+		}
+		if externalTimestamp > 0 {
+			return p.setEvaluatedValues(externalTimestamp, nil, nil)
```

Comment:
The second parameter is infoschema it should use. I think it should not be nil

Reply:
Fixed.